mllib/src/main/scala/org/apache/spark/ml/feature/StringIndexer.scala

```diff
@@ -146,7 +146,7 @@ object StringIndexer extends DefaultParamsReadable[StringIndexer] {
  * This is a temporary fix for the case when target labels do not exist during prediction.
  */
 @Since("1.4.0")
-class StringIndexerModel (
+class StringIndexerModel private[ml] (
```

Comment:
We make this public by mistake, so I think we should fix this bug. AFAIK, this is the only model whose constructor is public, so there is little chance to break user code. And I think it makes sense to break user code by fixing bug. Thanks.